test-suite/src/main/scala/testsuite/core/IsInstanceOfTest.scala

```diff
@@ -0,0 +1,59 @@
+package testsuite.core
+
+import testsuite.Assert.ok
+
+object IsInstanceOfTest {
+  def main(): Unit = {
+    ok(testInheritance())
+    ok(testMixinAll(new Child()))
+    ok(testMixinAll(new Parent()))
+    ok(testMixinAll(new Base {}))
+    ok(testMixin())
+    ok(testPrimitiveIsInstanceOfBase(5))
+    ok(testPrimitiveIsInstanceOfBase("foo"))
+
+    ok(!testInt("foo"))
+    ok(!testInt(2147483648L))
+    ok(testInt(3))
+    ok(!testInt(new Child()))
+    ok(testString("foo"))
+    ok(!testString(new Child()))
+  }
+
+  private def testInheritance(): Boolean = {
+    val child = new Child()
+    val parent = new Parent()
+    child.isInstanceOf[Parent] && child.isInstanceOf[Child] &&
+    parent.isInstanceOf[Parent] && !parent.isInstanceOf[Child]
+  }
+
+  private def testMixinAll(o: Base): Boolean = {
+    o.isInstanceOf[Base] && o.isInstanceOf[Base1] & o.isInstanceOf[Base2]
```

Comment:
I guess the `&` was not intentional?
```suggestion
    o.isInstanceOf[Base] && o.isInstanceOf[Base1] && o.isInstanceOf[Base2]
```